### PR TITLE
[DEV-2690] Card payment failure screen

### PR DESCRIPTION
**Pull request by lapatric**

### CODE REVIEW

File: src/screens/buy-failure.screen.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      <StyledVerticalStack gap={4}>
+        <div className="mx-auto">
+          <DfxIcon size={IconSize.XXL} icon={IconVariant.PROCESS_FAILED} color={IconColor.BLUE} />
+        </div>
+        <p className="text-base font-bold text-center text-dfxBlue-800">{header}</p>
+        <StyledButton
+          label={translate('screens/payment', 'Try again')}
+          onClick={() => navigate('/buy')}
+          width={StyledButtonWidth.FULL}
+        />
```
Comment (davidleomay) [inline src/screens/buy-failure.screen.tsx:30]: Reuse `Retry` translation from `general/actions`

File: src/screens/buy-failure.screen.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import { useNavigation } from '../hooks/navigation.hook';
+
+export function BuyFailureScreen(): JSX.Element {
+  const { translate } = useSettingsContext();
+  const { navigate } = useNavigation();
+  const header = translate('screens/payment', 'Your payment has failed. Please try again.');
+
+  return (
+    <Layout title={translate('screens/payment', 'Credit Card Payment')} backButton={false} textStart>
+      <StyledVerticalStack gap={4}>
```
Comment (davidleomay) [inline src/screens/buy-failure.screen.tsx:21]: Maybe use `Failed!` as a title? 😆 Would be the equivalent to the `Done!` on success:
image

File: src/components/exchange-rate.tsx [hunk 3]
```diff
@@ -1,7 +1,6 @@
-import { Asset, Blockchain, Fiat, Utils, useFiat } from '@dfx.swiss/react';
+import { Asset, Blockchain, Fiat, PriceStep, Utils, useFiat } from '@dfx.swiss/react';
 import { IconColor, StyledCollapsible, StyledInfoText, StyledVerticalStack } from '@dfx.swiss/react-components';
 import { Fees } from '@dfx.swiss/react/dist/definitions/fees';
-import { PriceStep } from '@dfx.swiss/react/dist/definitions/price-step';
```
Comment (davidleomay) [inline src/components/exchange-rate.tsx:4]: Good 👍 😉

File: src/translations/languages/de.json [hunk 4]
```diff
@@ -294,6 +294,8 @@
     "DFX fee": "DFX-Gebühr",
     "Network fee": "Netzwerk-Gebühr",
 
+    "Your payment has failed. Please try again.": "Ihre Zahlung ist fehlgeschlagen. Bitte versuchen Sie es erneut.",
+
```
Comment (davidleomay) [inline src/translations/languages/de.json:298]: In German we use `Du, not `Sie`

File: src/screens/buy-failure.screen.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import { useNavigation } from '../hooks/navigation.hook';
+
+export function BuyFailureScreen(): JSX.Element {
+  const { translate } = useSettingsContext();
+  const { navigate } = useNavigation();
+  const header = translate('screens/payment', 'Your payment has failed. Please try again.');
+
+  return (
+    <Layout title={`${translate('screens/payment', 'Failed')}!`} backButton={false} textStart>
+      <StyledVerticalStack gap={4}>
```
Comment (davidleomay) [inline src/screens/buy-failure.screen.tsx:21]: Missing translations?

Comment (lapatric) [reply src/screens/buy-failure.screen.tsx:21]: I don't think so, "Failed" is already contained in the translations or what do you mean?

Comment (davidleomay) [reply src/screens/buy-failure.screen.tsx:21]: Ok, true 😉

**Review by davidleomay — APPROVED** [review]

**Pull request merged**